xdsl/transforms/eqsat_create_eclasses.py

```diff
@@ -1,32 +1,91 @@
-from xdsl.dialects import eqsat, func
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, eqsat, func
 from xdsl.ir import Block
+from xdsl.passes import ModulePass
 from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
     PatternRewriter,
+    PatternRewriteWalker,
     RewritePattern,
     op_type_rewrite_pattern,
 )
 from xdsl.rewriter import InsertPoint, Rewriter
-from xdsl.utils.exceptions import DiagnosticException
 
 
 def insert_eclass_ops(block: Block):
+    # Insert eqsat.eclass for each operation
     for op in block.ops:
         results = op.results
         if len(results) != 1:
-            raise DiagnosticException("Ops with non-single results not handled")
+            continue
+            # TODO: ignore operations with no result for now
+            # raise DiagnosticException("Ops with non-single results not handled")
 
         eclass_op = eqsat.EClassOp(results[0])
-        Rewriter.insert_op(op, InsertPoint.after(eclass_op))
-        eclass_op.results[0].replace_by(eclass_op.result)
+        insertion_point = InsertPoint.after(op)
+        Rewriter.insert_op(eclass_op, insertion_point)
+        results[0].replace_by(eclass_op.results[0])
+        # Redirect eclassop operand back to the original value
+        # TODO: do we need a `replace_by_except` function, e.g.
+        # eclass_op.result.replace_by_except(eclass_op.results[0], [eclass_op.result])
```

Comment:
I'm not sure what MLIR has for this, but this works ok, no?

Reply:
I am thinking of this: https://mlir.llvm.org/doxygen/classmlir_1_1Operation.html#a6f5567fd62feeaf0cbba4877732c512a